### PR TITLE
prometheus-node-exporter-lua: add node_hwmon_temp_celsius metric

### PR DESCRIPTION
**Pull request by 0x5e**
Maintainer: @mweinelt 
Compile tested: TODO
Run tested: OpenWrt R21.11.11 on FriendlyElec NanoPi R2S

Description:
add `node_hwmon_temp_celsius` metric, the value can be found from `/sys/class/thermal/thermal_zone0/hwmon*/temp*_input`, chip name from `/sys/class/thermal/thermal_zone0/hwmon*/name`, sensor is `temp*`.

Output:
```
# TYPE node_hwmon_temp_celsius gauge
node_hwmon_temp_celsius{sensor="temp1",chip="soc_thermal"} 45.454
node_scrape_collector_duration_seconds{collector="hwmon"} 0.00026392936706543
node_scrape_collector_success{collector="hwmon"} 1
```

### CODE REVIEW

File: utils/prometheus-node-exporter-lua/Makefile [hunk 1]
```diff
@@ -44,6 +44,7 @@ define Package/prometheus-node-exporter-lua/install
 	$(INSTALL_BIN) ./files/usr/lib/lua/prometheus-collectors/time.lua        $(1)/usr/lib/lua/prometheus-collectors/
 	$(INSTALL_BIN) ./files/usr/lib/lua/prometheus-collectors/uname.lua       $(1)/usr/lib/lua/prometheus-collectors/
 	$(INSTALL_BIN) ./files/usr/lib/lua/prometheus-collectors/netclass.lua    $(1)/usr/lib/lua/prometheus-collectors/
+  $(INSTALL_BIN) ./files/usr/lib/lua/prometheus-collectors/hwmon.lua       $(1)/usr/lib/lua/prometheus-collectors/
```
Comment (champtar) [inline utils/prometheus-node-exporter-lua/Makefile:47]: I believe you are mixing spaces and tabs here

File: utils/prometheus-node-exporter-lua/files/usr/lib/lua/prometheus-collectors/hwmon.lua [hunk 2]
```diff
@@ -0,0 +1,15 @@
+local fs = require "nixio.fs"
+
+local function scrape()
+  local temp_metric = metric("node_hwmon_temp_celsius", "gauge")
+  for dir in fs.glob("/sys/class/thermal/thermal_zone0/hwmon*") do
```
Comment (champtar) [inline utils/prometheus-node-exporter-lua/files/usr/lib/lua/prometheus-collectors/hwmon.lua:5]: On my Turris Omnia, `/sys/class/hwmon/hwmon*/temp*_input` gives me 8 different sensors
```
# cat /sys/class/hwmon/hwmon*/name
f1072004mdiomii01
f10e4078.thermal
mv88e6xxx100
mv88e6xxx101
mv88e6xxx102
mv88e6xxx103
mv88e6xxx104
ath10k_hwmon
```
Why limit ourselves to 1 ?

**Review by champtar — CHANGES_REQUESTED** [review]
Can you confirm we get the same output as the official node exporter https://github.com/prometheus/node_exporter/blob/master/collector/hwmon_linux.go

**Comment by 0x5e** [comment]
This is the output from docker prom/node-exporter running on Synology NAS DS220+:
```
# HELP node_hwmon_temp_celsius Hardware monitor for temperature (input)
# TYPE node_hwmon_temp_celsius gauge
node_hwmon_temp_celsius{chip="platform_coretemp_0",sensor="temp1"} 46
node_hwmon_temp_celsius{chip="platform_coretemp_0",sensor="temp2"} 46
node_hwmon_temp_celsius{chip="platform_coretemp_0",sensor="temp3"} 46
```

This is the output from prometheus-node-exporter running on Debian 10:
```
# HELP node_hwmon_temp_celsius Hardware monitor for temperature (input)
# TYPE node_hwmon_temp_celsius gauge
node_hwmon_temp_celsius{chip="platform_iio_hwmon",sensor="temp1"} 35.937
```

The key and value are same, chip name seems different. I will update later.
I didn't implement all the hwmon metrics :)

**Comment by mweinelt** [comment]
Sorry for not getting back to you, but I don't think I'm the maintainer for this package. Nevertheless it would be great to have this!

**Comment by 0x5e** [comment]
Nevermind, I'm using docker container prom/node-exporter instead, it works great :)

**Pull request closed without merging**